hub.go

```diff
@@ -45,14 +47,47 @@ type hub struct {
 
 	// Unregister requests from connections.
 	unregister chan *connection
+
+	// Serial hub to communicate with serial ports
+	serialHub *serialhub
+
+	serialPortList *serialPortList
+
+	tools *tools.Tools
+
+	systray *systray.Systray
 }
 
-var h = hub{
-	broadcast:    make(chan []byte, 1000),
-	broadcastSys: make(chan []byte, 1000),
-	register:     make(chan *connection),
-	unregister:   make(chan *connection),
-	connections:  make(map[*connection]bool),
+func newHub(serialhub *serialhub, serialList *serialPortList, tools *tools.Tools, systray *systray.Systray) *hub {
+	hub := &hub{
+		broadcast:      make(chan []byte, 1000),
+		broadcastSys:   make(chan []byte, 1000),
+		register:       make(chan *connection),
+		unregister:     make(chan *connection),
+		connections:    make(map[*connection]bool),
+		serialHub:      serialhub,
+		serialPortList: serialList,
+		tools:          tools,
+		systray:        systray,
+	}
+
+	hub.serialHub.OnRegister = func(port *serport) {
```

Comment:
AFAIK `serialHub` is an internal dependency of the `hub` I don't think it is used else where. I think you should create it here directly